packages/axelar-wasm-std/src/voting.rs

```diff
@@ -251,25 +268,24 @@ impl Poll for WeightedPoll {
             return Err(Error::InvalidVoteSize);
         }
 
-        if participation.voted {
+        if participation.vote.is_some() {
             return Err(Error::AlreadyVoted);
         }
 
-        if self.status != PollStatus::InProgress {
-            return Err(Error::PollNotInProgress);
+        // late votes are not tallied
+        if self.status == PollStatus::InProgress {
+            self.tallies
+                .iter_mut()
+                .zip(votes.iter())
+                .filter(|(_, vote)| **vote)
+                .for_each(|(tally, _)| {
+                    *tally += Uint256::from(participation.weight);
+                });
         }
 
-        participation.voted = true;
-
-        self.votes
-            .iter_mut()
-            .zip(votes.into_iter())
-            .filter(|(_, vote)| *vote)
-            .for_each(|(tally, _)| {
-                *tally += Uint256::from(participation.weight);
-            });
+        participation.vote = Some(votes);
 
-        Ok(PollStatus::InProgress)
+        Ok(())
```

Comment:
```suggestion
        // late votes are not tallied
        if self.status != PollStatus::InProgress {
          return Ok(())
        }

        self.tallies
            .iter_mut()
            .zip(votes.iter())
            .filter(|(_, vote)| **vote)
            .for_each(|(tally, _)| {
                *tally += Uint256::from(participation.weight);
            });

        participation.vote = Some(votes);

        Ok(())
```
Exit early and leave the complicated code at the least indented place.